src/cts/src/LevelBalancer.cpp

```diff
@@ -94,22 +94,26 @@ void LevelBalancer::addBufferLevels(TreeBuilder* builder,
     totalX += clockInstObj->getX();
     totalY += clockInstObj->getY();
   }
-  const double centroidX = totalX / cluster.size();
+
   const double centroidY = totalY / cluster.size();
-  const int driverX = prevLevelSubNet->getDriver()->getX();
-  const int driverY = prevLevelSubNet->getDriver()->getY();
+  int x = prevLevelSubNet->getDriver()->getX();
+  int y = prevLevelSubNet->getDriver()->getY();
+
+  int dir = (centroidY - y) / std::abs(centroidY - y);
```

Comment:
Why use the same direction for both x & y?  You might be moving away from the centroid in x.

Reply:
Yes, I missed that. It was already fixed, thanks.